runtime/queries/nix/injections.scm

```diff
@@ -0,0 +1,48 @@
+; mark arbitary languages with a comment
+(
+  ((comment) @injection.language) . [
+    (indented_string_expression (_) @injection.content)
+    (_ (_)* (indented_string_expression (_) @injection.content))
+    (_ (_)* (_ (_)* (indented_string_expression (_) @injection.content)))
+    (_ (_)* (_ (_)* (_ (_)* (indented_string_expression (_) @injection.content))))
+    (_ (_)* (_ (_)* (_ (_)* (_ (_)* (indented_string_expression (_) @injection.content)))))])
+
+(binding
+  attrpath: (attrpath (identifier) @_path
+    (#match? @_path "(^\\w*Phase|(pre|post)\\w*|(.*\\.)?\\w*([sS]cript|[hH]ook)|(.*\\.)?startup)$"))
+  expression: [
+    (indented_string_expression (string_fragment) @injection.content
+      (#set! injection.language "bash"))
+    (_ (_)* (indented_string_expression (string_fragment) @injection.content
+      (#set! injection.language "bash")))
+    (_ (_)* (_ (_)* (indented_string_expression (string_fragment) @injection.content
+      (#set! injection.language "bash"))))])
```

Comment:
Properties (`#match?`, `#set!`, etc) apply to the entire match - it doesn't matter where in the match they're specified. So style-wise, stanzas tend to put them at the bottom at the outer-most level of nesting. This one would be:

```suggestion
((binding
   attrpath: (attrpath (identifier) @_path)
   expression: [
     (indented_string_expression (string_fragment) @injection.content)
     (_ (_)* (indented_string_expression (string_fragment) @injection.content))
     (_ (_)* (_ (_)* (indented_string_expression (string_fragment) @injection.content)))
   ])
 (#match? @_path "(^\\w*Phase|(pre|post)\\w*|(.*\\.)?\\w*([sS]cript|[hH]ook)|(.*\\.)?startup)$")
 (#set! injection.language "bash"))
```